### PR TITLE
Extract SkeletonLoader to @mattermost/components

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Skeleton loading component is now starting to be used newer designs of product. So i thought it would be a good idea to extract the component to package '@mattermost/components' so other products can use it as well. I have tried to keep the API of the component as simple and generic as possible to be used easily in other plugins/products.

#### Ticket Link
OSF

#### Related Pull Requests
n/a

#### Screenshots
No changes

#### Release Note

```release-note
SkeletonLoader components added in '@mattermost/components'
```

### CODE REVIEW

**Review by crspeller — APPROVED** [review]
Awesome!

File: components/activity_and_insights/insights/top_reactions/top_reactions.tsx [hunk 1]
```diff
@@ -64,18 +66,23 @@ const TopReactions = (props: WidgetHocProps) => {
...
                 <div
                     className='bar-chart-entry'
                     key={i}
                 >
-                    <BarChartLoader/>
-                    <CircleLoader
-                        size={20}
+                    <RectangleSkeletonLoader
+                        width='8px'
+                        height={`${barChartHeights[i]}px`}
```
Comment (hmhealey) [inline components/activity_and_insights/insights/top_reactions/top_reactions.tsx:81]: Perhaps we could make these optionally take a number and have number arguments get treated as pixel dimensions? I think React might do something similar already when using the `style` prop.

**Review by hmhealey — APPROVED** [review]
Looks great! I have one thought about possibly making the width/height parameters a bit easier to use, but this looks good to me

**Comment by hmhealey** [comment]
This has 2 dev reviews already, but I think that's make sure that someone from Suite Users (like Martin) reviews this so that they have visibility into it

File: components/activity_and_insights/insights/top_reactions/top_reactions.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply components/activity_and_insights/insights/top_reactions/top_reactions.tsx]: i have added that change as well https://github.com/mattermost/mattermost-webapp/pull/11011/commits/92a6b3ade26d37e511f874b9cf0aeb4c4ad3f6b7

**Comment by M-ZubairAhmed** [comment]
@mkraft i wasnt able to search anna.pohorielova

File: components/activity_and_insights/insights/top_reactions/top_reactions.tsx (hunk 1 above)
Comment (hmhealey) [reply components/activity_and_insights/insights/top_reactions/top_reactions.tsx]: Awesome, thanks

**Comment by M-ZubairAhmed** [comment]
@furqanmlk dont we need feature flag for insight for it run e2e? just wondering

**Comment by furqanmlk** [comment]
> @furqanmlk dont we need feature flag for insight for it run e2e? just wondering

It is enabled by default

**Review by furqanmlk — APPROVED** [review]
E2E passed

**Comment by mkraft** [comment]
cc @BenCookie95 FYI.

**Review by BenCookie95 — APPROVED** [review]
LGTM, nice work!

**Review by matthewbirtch — APPROVED** [review]
Seems to work as expected in the insights view. Nice work! I don't think it's used anywhere else yet though, right @M-ZubairAhmed?

**Pull request merged**